cmake/setup_cached_variables.cmake

```diff
@@ -165,6 +165,11 @@ OPTION(DEAL_II_UNITY_BUILD
   OFF)
 MARK_AS_ADVANCED(DEAL_II_UNITY_BUILD)
 
+OPTION(DEAL_II_EARLY_DEPRECATIONS
+  "Enable deprecation warnings for features deprecated since the last release."
+  ON)
+MARK_AS_ADVANCED(DEAL_II_EARLY_DEPRECATIONS)
```

Comment:
I am not sure I understand how this works. Doesn't this effectively make no difference between DEAL_II_DEPRECATED_EARLY and DEAL_II_DEPRECATED as the default is ON?

Reply:
Yes. We can debate about the default, of course. For me, it makes sense to avoid using deprecated code when developing deal.II by default and possibly disable the option (explicitly) if I want to use it in applications and don't want to see new deprecations.

Reply:
Indeed this is a point we should discuss. Should the default value be `OFF` or `ON`? I would be in favor of `ON`.

Reply:
I like OFF a bit more. As discussed in the other PR, we developers know how to enable it and all CI machines use it, but we let other users have a more gentle warning message.

Reply:
It's now disabled by default.